### PR TITLE
W-13776337 Review "Implement GraphQL API" docs

### PR DESCRIPTION
**Pull request by chris-campbell9**

### CODE REVIEW

File: modules/ROOT/pages/configure-graphql-flow-responses.adoc [hunk 1]
```diff
@@ -34,13 +34,28 @@ See xref:implement-a-graphql-api.adoc[Implement a GraphQL API].
...
-. Modify the `<set-payload>` component with the following JSON response:
+. Replace the `<set-payload>` component with the following JSON response:
 +
 [source,XML]
 --
 <set-payload value='{"id": 1, "name": "My Book", "pageCount": 28, "author": "author-1"}' mimeType="application/json" />
 --
-. Relaunch your application and send a `bookById` query:
+. Select Run and Debug in the Activity Bar, then press the *Start Debugging* icon.
+. In your preferred REST client, create a `POST` request to the local address and send a `bookById` query:
```
Comment (dukesphere) [inline modules/ROOT/pages/configure-graphql-flow-responses.adoc]: @chris-campbell9 What query did you use?

File: modules/ROOT/pages/test-graphql-first-implementation.adoc [hunk 2]
```diff
@@ -57,15 +61,16 @@ image::ports-panel-local-address-8081.png[]
...
+. In your preferred REST client, create a `POST` request to the local address, adding the `/graphql` path: `+https://<web-IDE-instance>/proxy/8081/graphql+`.
 . Add a new *Cookie* header to the request with the `proxysession` value:
 +
 [source,CURL]
 --
 --header 'Cookie: proxysession5f5648c7115f4e2e8f3f9396355ac4e8-0=xxxxxxxxxxxxxx \
 --
 +
-image::send-request-with-cookies.png[]
+// TODO: cc: We probably shouldn't have screenshots of third-party applications, due to maintenance issues.
```
Comment (dukesphere) [inline modules/ROOT/pages/test-graphql-first-implementation.adoc]: [agree](agree: Let's leave it commented out for now. I'll create a story to address all TODOs in the beta docs.)

File: modules/ROOT/pages/test-graphql-first-implementation.adoc [hunk 3]
```diff
@@ -93,7 +98,8 @@ Note that because you are querying a value by a specific ID (`$id: ID`), you mus
 }
 --
 +
-image::graphql-query-book-by-id.png[]
+// TODO: cc: We probably shouldn't have screenshots of third-party applications, due to maintenance issues.
+// image::graphql-query-book-by-id.png[]
```
Comment (dukesphere) [inline modules/ROOT/pages/test-graphql-first-implementation.adoc:119]: agree: Let's leave it commented out for now. I'll create a story to address all TODOs in the beta docs.

File: modules/ROOT/pages/implement-a-graphql-api.adoc [hunk 4]
```diff
@@ -47,15 +52,17 @@ image::implement-api-configure-in-task.png[]
 .. *Project Name*: _books-implementation_.
 .. *Project Location*: select *Browse*, and then select your home directory.
 +
-image::select-home-directory.png[]
+//TODO: cc: I wonder if we need this screenshot? The step seems self-explantory without one. Plus, the screenshot differs from the Desktop IDE UI.
```
Comment (dukesphere) [inline modules/ROOT/pages/implement-a-graphql-api.adoc:57]: agree: Let's leave it commented out for now. I'll create a story to address all TODOs in the beta docs.

File: modules/ROOT/pages/configure-graphql-flow-responses.adoc (hunk 1 above)
Comment (chris-campbell9) [reply modules/ROOT/pages/configure-graphql-flow-responses.adoc]: @dukesphere I had to modify the query in the following way:

`query BookById { 
     bookById(id: "1") {
         id
         name
         author {
             id
             firstName
             lastName
         }
         pageCount
     }
}`

Comment (chris-campbell9) [reply modules/ROOT/pages/configure-graphql-flow-responses.adoc:44]: Basically I had to replace:
`query bookById($id: ID)`
With
`query bookById`

Comment (chris-campbell9) [reply modules/ROOT/pages/configure-graphql-flow-responses.adoc]: Basically I had to replace:
`query bookById($id: ID)`
With
`query bookById`

Comment (dcourtil) [reply modules/ROOT/pages/configure-graphql-flow-responses.adoc]: The problem with the original query from the documentation is that it is missing the variable value. You have to provide a JSON with each key-value in order for a query with arguments to be valid.
This is, for providing a value to the variable `$id`, you have to add this JSON the the graphql variables field
```
{
    "id": 1
}
```
Depending on the (REST) client, some (as Postman) have a special box for setting graphql variables. If the client does not have a special UI for triggering graphql requests, then you have to just run a POST with a JSON containing both the query and the variables, as following:
```
{
   "query":"query bookById($id: ID) {bookById(id: $id){ id name pageCount author{ id firstName lastName}}}",
   "variables":{
      "id":1
   }
}
```

Comment (dukesphere) [reply modules/ROOT/pages/configure-graphql-flow-responses.adoc]: @chris-campbell9 Can you take a look at ☝️ when you have a moment?

Comment (chris-campbell9) [reply modules/ROOT/pages/configure-graphql-flow-responses.adoc]: Thanks @dcourtil. 

In Postman, I could only get things to work when I used the following query:

```
query BookById {
    bookById(id: "1") {
        id
        name
        author {
            id
            firstName
            lastName
        }
        pageCount
    }
}
```

I did enter the graphql variable value in a special box UI, and the above is the query that was generated.

Comment (chris-campbell9) [reply modules/ROOT/pages/configure-graphql-flow-responses.adoc]: Nevermind, using `curl` I was able to use what you mentioned.

File: modules/ROOT/pages/configure-graphql-flow-responses.adoc [hunk 5]
```diff
@@ -34,30 +36,17 @@ See xref:implement-a-graphql-api.adoc[Implement a GraphQL API].
...
 --
-. Modify the `<set-payload>` component with the following JSON response:
+. Replace the `<set-payload>` component with the following JSON response:
 +
 [source,XML]
 --
 <set-payload value='{"id": 1, "name": "My Book", "pageCount": 28, "author": "author-1"}' mimeType="application/json" />
 --
-. Relaunch your application and send a `bookById` query:
+. Select Run and Debug in the Activity Bar, then press the *Start Debugging* icon.
```
Comment (dmerlob) [inline modules/ROOT/pages/configure-graphql-flow-responses.adoc]:
```suggestion
. Select *Run and Debug* in the Activity Bar, then press the *Start Debugging* icon.
```

File: modules/ROOT/pages/test-graphql-first-implementation.adoc [hunk 6]
```diff
@@ -1,34 +1,40 @@
...
 
-. Launch Anypoint Code Builder.
-. Open the *Books Implementation* project.
-. On your Run and Debug view, select your *Run and Debug* icon.
+.Web IDE
+[%collapsible]
+====
+
+. In Anypoint Code Builder, open the *Books Implementation* project.
+. Select Run and Debug in the Activity Bar, then press the *Start Debugging* icon.
```
Comment (dmerlob) [inline modules/ROOT/pages/test-graphql-first-implementation.adoc]:
```suggestion
. Select *Run and Debug* in the Activity Bar, then press the *Start Debugging* icon.
```

File: modules/ROOT/pages/test-graphql-first-implementation.adoc [hunk 7]
```diff
@@ -57,16 +63,32 @@ image::ports-panel-local-address-8081.png[]
...
+        "id": 1
+    }
+}'
 --
 +
-image::send-request-with-cookies.png[]
-. Configure the body of the request to contain a GraphQL query. For example:
+// . Add a new *Cookie* header to the request with the `proxysession` value:
+// +
+// [source,CURL]
```
Comment (dmerlob) [inline modules/ROOT/pages/test-graphql-first-implementation.adoc]:
```suggestion
// [source,curl]
```

File: modules/ROOT/pages/test-graphql-first-implementation.adoc [hunk 8]
```diff
@@ -93,9 +115,46 @@ Note that because you are querying a value by a specific ID (`$id: ID`), you mus
...
+}
+--
+====
+
+.Desktop IDE
+[%collapsible]
+====
+
+. In Anypoint Code Builder, open the *Books Implementation* project.
+. Select Run and Debug in the Activity Bar, then press the *Start Debugging* icon.
```
Comment (dmerlob) [inline modules/ROOT/pages/test-graphql-first-implementation.adoc]:
```suggestion
. Select *Run and Debug* in the Activity Bar, then press the *Start Debugging* icon.
```

**Review by dmerlob — APPROVED** [review]

**Pull request merged**